packages/@angular/cli/blueprints/ng/index.ts

```diff
@@ -78,6 +91,17 @@ export default Blueprint.extend({
       fileList = fileList.filter(p => p.indexOf('app.component.spec.ts') < 0);
     }
 
+    if (this.options && this.options.barebones) {
+      fileList = fileList.filter(p => p.indexOf('e2e/') < 0);
+      fileList = fileList.filter(p => p.indexOf('editorconfig') < 0);
+      fileList = fileList.filter(p => p.indexOf('README.md') < 0);
+      fileList = fileList.filter(p => p.indexOf('karma.conf.js') < 0);
+      fileList = fileList.filter(p => p.indexOf('protractor.conf.js') < 0);
+      fileList = fileList.filter(p => p.indexOf('test.ts') < 0);
+      fileList = fileList.filter(p => p.indexOf('tsconfig.spec.json') < 0);
+      fileList = fileList.filter(p => p.indexOf('tslint.json') < 0);
+    }
```

Comment:
Another way is to save all those files in a list, and use `fileList = fileList.filter(p => blackList.indexOf(p) < 0);`